frame/bribedao/src/sortedvec.rs

```diff
@@ -0,0 +1,523 @@
+            ///
+            /// # Panics
+            ///
+            /// Panics if `at > len`.
+            pub fn split_off(&mut self, at: usize) -> Self {
+                let other_inner = self.inner.split_off(at);
+                Self {
+                    inner: other_inner,
+                }
+            }
+
+            /// Removes all elements but one that resolve to the same key.
+            pub fn dedup(&mut self) {
+                self.inner.dedup_by(|a, b| Self::derive_key(a) == Self::derive_key(b));
+            }
+
+            /// Removes and returns the greatest element with the respect to
+            /// the generated keys. An `O(1)` operation.
+            pub fn pop(&mut self) -> Option<$val> {
+                self.inner.pop()
+            }
+
+            // private method
+            fn sort(&mut self) {
+                self.inner.sort_unstable_by(|a, b| {
+                    let lhs = Self::derive_key(a);
+                    let rhs = Self::derive_key(b);
+                    lhs.cmp(&rhs)
+                })
+            }
+        }
+
+        impl Into<Vec<$val>> for $name {
+            fn into(self) -> Vec<$val> {
+                self.inner
+            }
+        }
+
+        impl std::default::Default for $name {
+            fn default() -> Self {
+                Self { inner: std::default::Default::default() }
+            }
+        }
+
+        impl Extend<$val> for $name {
+            fn extend<I>(&mut self, iter: I)
+            where
+                I: IntoIterator<Item = $val>,
+            {
+                self.inner.extend(iter);
+                self.sort();
+            }
+        }
+
+        impl std::iter::FromIterator<$val> for $name {
+            fn from_iter<I: std::iter::IntoIterator<Item=$val>>(iter: I) -> Self {
+                let inner = Vec::from_iter(iter);
+                From::from(inner)
+            }
+        }
+
+        impl From<Vec<$val>> for $name {
+            fn from(vec: Vec<$val>) -> Self {
+                let mut res = Self { inner: vec };
+                res.sort();
+                res
+            }
+        }
+
+        impl std::ops::Deref for $name {
+            type Target = Vec<$val>;
+
+            fn deref(&self) -> &Self::Target {
+                &self.inner
+            }
+        }
+
+        impl std::borrow::Borrow<[$val]> for $name {
+            fn borrow(&self) -> &[$val] {
+                &self.inner
+            }
+        }
+
+        impl AsRef<[$val]> for $name {
+            fn as_ref(&self) -> &[$val] {
+                &self.inner
+            }
+        }
+
+        impl AsRef<Vec<$val>> for $name {
+            fn as_ref(&self) -> &Vec<$val> {
+                &self.inner
+            }
+        }
+    }
+}
+
+#[derive(Debug, Default, TypeInfo, PartialEq, Hash, Clone, Copy, Encode, Decode)]
+pub struct BribesStorage {
+	pub p_id: u32,
+	pub amount: u32,
+	pub votes: u32,
+}
```

Comment:
We do we have this struct defined in 3 different files?

Reply:
used in removed tests files